x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/api/engines/create_engine_api_logic.ts

```diff
@@ -14,17 +14,17 @@ export interface CreateEngineApiParams {
   indices: string[];
 }
 
-export type CreateEngineApiResponse = EnterpriseSearchEngine;
+export type CreateEngineApiResponse = EnterpriseSearchEngineUpsertResponse;
 
 export type CreateEngineApiLogicActions = Actions<CreateEngineApiParams, CreateEngineApiResponse>;
 
 export const createEngine = async ({
   engineName,
   indices,
 }: CreateEngineApiParams): Promise<CreateEngineApiResponse> => {
-  const route = `/internal/enterprise_search/engines`;
+  const route = `/internal/enterprise_search/engines/${engineName}`;
```

Comment:
so we will need a `create` flag here to make sure the call doesn't "update" an existing search application if the name used already exists. If exists, the API should throw an exception here.

example "PUT _application/search_application/my-app?create"

cc @kderusso to validate my comments

Reply:
Yes, without `create` the call will always work, even if the search application already exists.

Reply:
Thinking of logic here, So when creating an engine, we could add a logic to do a GET engine call and validate the response before actually making a PUT request?

Reply:
You just have to add the `create` flag to the PUT call. That will return a 409 if the engine already exists. Full docs can be found [here](https://github.com/elastic/elasticsearch/blob/ent-search-module/docs/reference/search-application/apis/put-search-application.asciidoc)